### ISSUE TITLE
After upgrade to stable-5142-2 not able to open meeting

### ISSUE THREAD

**Issue by menzbua** [jitsi/docker-jitsi-meet#854]
Hi,

after upgrading to stable-5142-2 i can not create a new meeting. After starting the meeting i got an error that the network connection was disconnected.

In the browser logs i see this error:
```
Firefox kann keine Verbindung zu dem Server unter wss://localhost:8443/xmpp-websocket?room=compellingnotebookswatersuccessfully aufbauen. strophe.umd.js:5463:30
020-12-06T07:33:28.375Z [JitsiMeetJS.js] <getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  Error: Strophe: Websocket error [object Event]
```

I have an nginx reverse proxy deployed. The config for the xmpp location is this:

    location /xmpp-websocket {
        proxy_pass              http://jitsi_prosody:5280/xmpp-websocket;
        proxy_http_version      1.1;
        proxy_set_header        Upgrade $http_upgrade;
        proxy_set_header        Connection "upgrade";
        #proxy_set_header        Host $host;
        tcp_nodelay             on;
    }

I've created my own docker network and my own names in this network for the frontend and the prosody. This is why the name is not default.

Does any one know what i have to do in the nginx config to get it up and running.

EDIT1:

I've deployed it with 5142 web version and it is working. It is not working when i deply with 5142-2 web version:

```
jitsi/jvb                        latest              2aee468f92f2        39 hours ago        279MB
jitsi/jicofo                     latest              a4aaf27f57fd        39 hours ago        276MB
jitsi/prosody                    latest              cc3b4779f63c        39 hours ago        118MB
jitsi/web                        stable-5142         4d1876107fae        7 weeks ago         487MB
```

Thanks a lot

Regards
Manuel

**Comment by DerTomm** [comment]
I also encountered problems after upgrading from stable-5142 to stable-5142-1. In my case I had to replace my former `.env` file with the new example template of release stable-5142-1 as some XMPP related default values where changed. After customizing the new env file and adding this xmpp-websocket section to my nginx config it now works:

```
location /xmpp-websocket {
        proxy_pass http://127.0.0.1:180/xmpp-websocket;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;
        tcp_nodelay on;
   }
```

(nginx redirects incoming websocket requests not to prosody as in your case but to the internal jitsi docker nginx:)
```
PORTS                                              NAMES
0.0.0.0:4443->4443/tcp, 0.0.0.0:10000->10000/udp   jitsi-docker_jvb_1
127.0.0.1:180->80/tcp, 127.0.0.1:1443->443/tcp     jitsi-docker_web_1     <-----
                                                   jitsi-docker_jicofo_1
5222/tcp, 5280/tcp, 5347/tcp                       jitsi-docker_prosody_1
```
Maybe this helps you to solve the problem.

**Comment by menzbua** [comment]
I've replaced the .env file with the latest one from the repo and redirected the xmpp-websocket to the correct container. But i see the same issue.

**Comment by DerTomm** [comment]
@menzbua Did you delete and recreate the jitsi configuration folder (and its subfolders) before restarting the containers?

**Comment by mfts** [comment]
@DerTomm thanks for preemptively sharing the NGINX config. Works like a charm for me

**Comment by pandiloko** [comment]
I'm using Apache as a proxy and wasn't able to get it to work with: 

```
    RewriteEngine on
    RewriteCond ${HTTP:Upgrade} websocket [NC]
    RewriteCond ${HTTP:Connection} upgrade [NC]
    RewriteRule .* "wss:/127.0.0.1:8000/$1" [P,L]

    ProxyPass        / http://127.0.0.1:8000/
    ProxyPassReverse / http://127.0.0.1:8000/
```
I couldn't spend more time on it as my colleagues wanted to use the service so I disabled the xmpp web socket with:

`ENABLE_XMPP_WEBSOCKET=0`

at the end of the .env file (also deleting ~/.jitsi-meet-cfg and allowing to regenerate). So far **it seems to work again.**

Someone cares to share a working snippet for Apache?

**Comment by nilsrb** [comment]
Same error, using now to get it working again jitsi/web:stable-5076

**Comment by menzbua** [comment]
No, i did not delete the config folder. I have some Users configured they would be also deleted when i'm deleting the config directory, correct?

Thanks
Regards
Manuel

**Comment by pandiloko** [comment]
It is working for me now. In case anyone out there uses Apache, the problem seems to be with the redirect to `wss://`. I do the encryption on Apache (`DISABLE_HTTPS=1` in `.env`) so the redirect should be to `ws://` instead. 

To summarize: Apache frontend, jitsi stable-5142-2 running in a Docker instance, DISABLE_HTTPS=1, and here the relevant part from Apache config file:

```
        SSLProxyEngine on
        Include options-ssl-apache.conf

        ProxyPreserveHost On
        RequestHeader set X-Forwarded-Proto "https"

        ProxyPass        / http://127.0.0.1:8000/
        ProxyPassReverse / http://127.0.0.1:8000/

        RewriteEngine on
        RewriteCond %{HTTP:Connection} Upgrade [NC]
        RewriteCond %{HTTP:Upgrade} websocket [NC]
        RewriteRule ^/?(.*) "ws://127.0.0.1:8000/$1" [P,L]
```

Hope it helps anyone.

**Comment by aeburriel** [comment]
For Apache please note proxy_wstunnel module has to be enebled.

**Comment by saghul** [comment]
-3 is now out which regenerates the prosody config file, this was one of the main problems when upgrading. Please test -3 and open a new issue of something is still not working. Changes are still necessary if a reverse proxy is used in front of this setup. https://github.com/jitsi/docker-jitsi-meet/blob/master/CHANGELOG.md#stable-5142-3

**Comment by maximberezin97** [comment]
@saghul Can you please elaborate on what changes need to be made for those running the Docker Compose Jitsi stack behind a reverse proxy?

I am using nginx based on the [linuxserver/swag](https://github.com/linuxserver/docker-swag) container. This is my `/xmpp-websocket` block, which has been working untouched in my config for 6+ months:
```
location /xmpp-websocket {
        include                 /config/nginx/proxy.conf;
        set                     $upstream_xmpp jitsi_xmpp;
        proxy_pass              http://$upstream_xmpp:5280/xmpp-websocket;
        proxy_set_header        Upgrade $http_upgrade;
        proxy_set_header        Connection "upgrade";
        proxy_set_header        Host $host;
        tcp_nodelay             on;
    }
```
where `/config/nginx/proxy.conf` is based on https://github.com/linuxserver/docker-swag/blob/master/root/defaults/proxy.conf

According to [DerTomm's solution above](https://github.com/jitsi/docker-jitsi-meet/issues/854#issuecomment-739503859), and guessing based on the [changelog](https://github.com/jitsi/docker-jitsi-meet/blob/master/CHANGELOG.md#stable-5142-3), it appears that the `/xmpp_websockets` path was rerouted to the jitsi/web container. Is that correct?
Implementing that reroute, even after reconfiguring my .env using the most recent env.example, does not solve the problem on my setup.

**Comment by laf0rge** [comment]
I am still experiencing this problem with stable-5142-4 behind a nginx reverse proxy. I evene deleted all the configs from ~/.jitsi-meet-cfg and started from scratch. The only work-around that makes jitsi meet functional again is to use ENABLE_XMPP_WEBSOCKET=0

**Comment by zougloub** [comment]
Did an upgrade, but I'm using `ENABLE_XMPP_WEBSOCKET=0` for now as I couldn't quickly figure out how to configure the reverse proxying.

Without ENABLE_XMPP_WEBSOCKET,:
- From `env.example` I have set `PUBLIC_URL`, `DOCKER_HOST_ADDRESS`, `ENABLE_LETSENCRYPT=0`, ` DISABLE_HTTPS=1` and the HTTP port was 8000
- I had set up an additional nginx rule for wss reverse forwarding, based on an above comment:
```
                location /xmpp-websocket {      
                        # Timeout if the real server is dead
                        proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;

                        # Proxy Connection Settings
                        proxy_buffers 32 4k;
                        proxy_connect_timeout 240;
                        proxy_headers_hash_bucket_size 128;
                        proxy_headers_hash_max_size 1024;
                        proxy_http_version 1.1;
                        proxy_read_timeout 240;  
                        proxy_redirect  http://  ws://;
                        #proxy_redirect  http://  $scheme://;
                        proxy_send_timeout 240;

                        # Proxy Cache and Cookie Settings
                        proxy_cache_bypass $cookie_session;
                        #proxy_cookie_path / "/; Secure"; # enable at your own risk, may break certain apps
                        proxy_no_cache $cookie_session;

                        # Proxy Header Settings
                        #proxy_set_header Connection $connection_upgrade;
                        proxy_set_header Early-Data $ssl_early_data;
                        proxy_set_header Host $host;
                        proxy_set_header Upgrade $http_upgrade;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_set_header X-Forwarded-Host $host;
                        proxy_set_header X-Forwarded-Proto https;
                        proxy_set_header X-Forwarded-Ssl on;
                        proxy_set_header X-Real-IP $remote_addr;

                        proxy_pass http://localhost:8000/xmpp-websocket;
                        proxy_set_header Connection "Upgrade";
                        tcp_nodelay             on;
                }     
```

And was getting 404's.

**Comment by aldenjenkins** [comment]
I solved this issue by properly setting the PUBLIC_URL environment variable with the proper `https://` protocol prefix (with no trailing slash)

**Comment by gerroon** [comment]
I disabled web socket (ENABLE_XMPP_WEBSOCKET=0 ) and using  @pandiloko Apache  lines, I still have this issue. It still complaints about wesocket and asking the user to rejoin.

I have set all these properly PUBLIC_URL, DOCKER_HOST_ADDRESS, ENABLE_LETSENCRYPT=0, DISABLE_HTTPS=1 No solution so far.

**Comment by pandiloko** [comment]
@gerroon I noticed on your screenshot the error: 

`WebSocket connection to 'wss://[...]' failed`

It seems it is still trying to connect to wss instead ws. I think that could be the problem. Are you sure you don't have any other Rewrite rules interfering?

**Comment by gerroon** [comment]
Yeah somehow passing these does not disable wss

```
ENABLE_SCTP=1
ENABLE_COLIBRI_WEBSOCKET=0
ENABLE_XMPP_WEBSOCKET=0

```

**Comment by maglub** [comment]
As @aldenjenkins mention above. Setting the PUBLIC_URL to a proper value with override the default "localhost".

In my case, where my docker host has the internal ip address 192.168.4.225, I just added this to the `.env` file:

```
PUBLIC_URL=https://192.168.4.225:8443
```

**Comment by the-lazy-fox** [comment]
Hello @maximberezin97 
Where you able to make it work for SWAG reverse proxy?
I still have to turn XMPP off 👎🏼 
Thank you!

**Comment by JiangLongLiu** [comment]
just [close the WebSocket](https://jitsi.github.io/handbook/docs/devops-guide/devops-guide-docker/#disabling-websocket-connections) in `docker-compose.yml` ，then run  `docker-compose up -d` again

edit `docker-compose.yml`  environment

``` yml
jicofo 
	- ENABLE_SCTP=1
jvb
	- ENABLE_COLIBRI_WEBSOCKET=0
web
	- ENABLE_COLIBRI_WEBSOCKET=0
prosody
	- ENABLE_XMPP_WEBSOCKET=0
	- ENABLE_XMPP_WEBSOCKET=0
```

**Comment by working-name** [comment]
Hello from the future. Yes, jitsi's .env is still as _great_ as it was in 2022. The culprit is this: `PUBLIC_URL`, which includes a trailing `:{HTTPS_PORT}` at the end of the URL. You need to remove that junk if you're proxying this. If you look at the connection attempts in developer tools, you'll see that wss:// is trying your public URL with the `8443` port appended...

This wasn't as explicit about it, but just **REMOVE :{HTTPS_PORT} from PUBLIC_URL**.
> I solved this issue by properly setting the PUBLIC_URL environment variable with the proper `https://` protocol prefix (with no trailing slash)

**Comment by the-lazy-fox** [comment]
I'm trying to set up Jitsi via Docker + SWAG again (3.5 years later), and I'm somehow ending up in the same thread with the same issue…

Thanks @JiangLongLiu for the old answer - the missing tag was the reason I overlooked it 😬
I applied your suggestion and the error changed from:
`Firefox can’t establish a connection to the server at wss://localhost:8443/xmpp-websocket`
(which drove me crazy since I don't even use that port)
to:
`Firefox can’t establish a connection to the server at wss://sub.mydomain.abc/xmpp-websocket`

So there’s some progress, but it still doesn’t work.

@maximberezin97 since you’re also using SWAG, did you ever get this working?

Thanks!
